pkg/nuclide-flow/lib/FlowAutocompleteProvider.js

```diff
@@ -134,9 +134,9 @@ function updateResults(
 }
 
 function getReplacementPrefix(originalPrefix: string): string {
-  // If it is just whitespace and punctuation, ignore it (this keeps us
-  // from eating leading dots).
-  return /^[\s.]*$/.test(originalPrefix) ? '' : originalPrefix;
+  // Ignore prefix unless it's an identifier (this keeps us form eating leading
+  // dots, colons, etc).
+  return /^[a-zA-Z0-9_$]*$/.test(originalPrefix) ? originalPrefix : '';
```

Comment:
I've inlined an identifier regex in `shouldFilter` in this same file -- could you pull that out into a constant and use it here? Notably, what you have here allows numbers at the beginning of an identifier which is incorrect (not a big issue in this case but why not do it right).